x-pack/plugins/security_solution/public/management/cypress/e2e/artifact_tabs_in_policy_details.cy.ts

```diff
@@ -0,0 +1,221 @@
+        });
+
+        it(`[ALL] User can add ${testData.title} artifact`, () => {
+          loginWithPrivilegeAll();
+          visitArtifactTab(testData.tabId);
+
+          cy.getBySel('policy-artifacts-empty-unexisting').should('exist');
+
+          cy.getBySel('unexisting-manage-artifacts-button').should('exist').click();
+
+          const { formActions, checkResults } = testData.create;
+
+          performUserActions(formActions);
+
+          // Add a per policy artifact - but not assign it to any policy
+          cy.get('[data-test-subj$="-perPolicy"]').click(); // test-subjects are generated in different formats, but all ends with -perPolicy
+          cy.getBySel(`${testData.pagePrefix}-flyout-submitButton`).click();
+
+          // Check new artifact is in the list
+          for (const checkResult of checkResults) {
+            cy.getBySel(checkResult.selector).should('have.text', checkResult.value);
+          }
+
+          cy.getBySel('policyDetailsPage').should('not.exist');
+          cy.getBySel('backToOrigin').contains(/^Back to .+ policy$/);
+
+          cy.getBySel('backToOrigin').click();
+          cy.getBySel('policyDetailsPage').should('exist');
+        });
+      });
+
+      context(`Given there are no assigned ${testData.title} entries`, () => {
+        it(`[READ] User CANNOT Manage or Assign ${testData.title} artifacts`, () => {
+          loginWithPrivilegeRead(testData.privilegePrefix);
+          visitArtifactTab(testData.tabId);
+
+          cy.getBySel('policy-artifacts-empty-unassigned').should('exist');
+
+          cy.getBySel('unassigned-manage-artifacts-button').should('not.exist');
+          cy.getBySel('unassigned-assign-artifacts-button').should('not.exist');
+        });
+
+        it(`[ALL] User can Manage and Assign ${testData.title} artifacts`, () => {
+          loginWithPrivilegeAll();
+          visitArtifactTab(testData.tabId);
+
+          cy.getBySel('policy-artifacts-empty-unassigned').should('exist');
+
+          // Manage artifacts
+          cy.getBySel('unassigned-manage-artifacts-button').should('exist').click();
+          cy.location('pathname').should(
+            'equal',
+            `/app/security/administration/${testData.urlPath}`
+          );
+          cy.getBySel('backToOrigin').click();
+
+          // Assign artifacts
+          cy.getBySel('unassigned-assign-artifacts-button').should('exist').click();
+
+          cy.getBySel('artifacts-assign-flyout').should('exist');
+          cy.getBySel('artifacts-assign-confirm-button').should('be.disabled');
+
+          cy.getBySel(`${testData.artifactName}_checkbox`).click();
+          cy.getBySel('artifacts-assign-confirm-button').click();
+        });
+      });
+
+      context(`Given there are assigned ${testData.title} entries`, () => {
+        it(`[READ] User can see ${testData.title} artifacts but CANNOT assign or remove from policy`, () => {
+          loginWithPrivilegeRead(testData.privilegePrefix);
```

Comment:
This looks very similar to the previous READ tests. Where are the entries being added? 🤔

Reply:
this is the trick @dasansol92 used in the artifact pages tests (#148404): the test cases reuse the state created by the previous test cases:
- `[ALL] User can add ${testData.title} artifact` adds the artifact but does not assign it to the current policy,
- `[ALL] User can Manage and Assign ${testData.title} artifacts` assigns the artifact.

I think this is an exciting question, because this way the tests are not really independent from each other, and this is usually an anti-pattern. On the other hand, the test cases that depend on each other are well contained, so they should have effect only on each other _inside_ the test group, as the `before` and `after` hooks clean up the artifacts, so in this case maybe we can allow this to make the test runs faster - but we will see in time I guess if this was a good idea or not.

Reply:
Good call! Yes, it seems an anti-pattern but I would say that make sense for e2e tests where there are API and other async interactions, so we keep the test run faster and reuse the data already created (when needed). Otherwise, we should create and clean artifacts on each `it()` and that would increment the test execution time a lot. 
I'm more than happy to discuss this further 🙂

Reply:
I would suggest that we contain conditions in each test, even if the tests take longer. One issue with dependent tests is debugging/running tests locally. If I want to run a single test that's dependent on another it would just fail. I made a similar refactoring to my set of tests as well to ensure that I have an artifact that I need for a test to verify whether I can update it or not. See [here](https://github.com/elastic/kibana/pull/148560#discussion_r1081322810)

Reply:
Yeah, that's a good point @ashokaditya - running locally all tests from engineer time is probably more expensive, than having slightly slower tests in CI time.

I'm happy to do the modification, and return with the results.


One additional benefit is that we can have only one user journey for the `ALL` access, because there is no need to keep the 'output' for the `READ` cases. On one hand, I think user journeys show nicely how the user interacts with the sw and also it would be faster, but on the other hand, having longer tests with no detailed descriptions may reduce readability. I'll prepare this one too, so we can discuss it.

Reply:
hey @ashokaditya @dasansol92 

I pushed the modifications for the independent test cases, there are 2 versions.
(For reference, the runtime for the previous solution on my laptop was 🕐 `3:00`.)

1) Small, independent test cases - 4f28c66
🕐 Runtime: `5:44`
One small (but not very small) `READ` and `ALL` test case for every scenario (no / unassigned / assigned artifact)
Slower, but more descriptive.
<img width="423" alt="image" src="https://user-images.githubusercontent.com/39014407/215474218-43af5436-f74a-4e97-b35a-46d5a690152b.png">

2) Minimal number of test cases - ec96e8f
🕐 Runtime: `4:36`
What can be done in one test case,  is done - i.e. the ALL privilege has a long user journey.
A bit faster, and testing user journey is the go-to suggestion.
<img width="430" alt="image" src="https://user-images.githubusercontent.com/39014407/215473692-bb3565c2-dad9-4abc-9267-c7a0a8f106f8.png">


Which one do you prefer?
For me, I think I prefer the first one, because it shows us better the difference between the privileges in the same scenarios, even if the overall runtime is slower. Also, the user journey looks a bit of an exaggeration, because it's probably not a realistic user journey when the user creates, then assigns, then unassigns artifacts in one breath. But I am curious about your opinions!

Reply:
I would go for the first one too, more descriptive and the time execution difference is not huge. Thanks for doing this analysis, super helpful!

Reply:
Absolutely agree. Let's go with the 1. 🚀